### PR TITLE
Fix issues in THistPainter and TFileMerger

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
To protect against a division by zero? The original patch is from @pzhristov, so he might have more to say.

**Comment by couet** [comment]
Yes very likely. I was just wondering if there was some kind of reproducer for a specific situation. It does not arm anyway.

**Comment by ktf** [comment]
```
Double_t dz = zmax - zmin;
```

I guess an histogram with a single bin?

**Comment by couet** [comment]
It seems already protected :
```
root [0] auto h = new TH2D("h","h",1,0,1,1,0,1);
root [1] h->Draw("COL")
Info in <TCanvas::MakeDefCanvas>:  created default TCanvas with name c1
root [2] 
```

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: io/io/src/TFileMerger.cxx [hunk 1]
```diff
@@ -776,8 +776,10 @@ Bool_t TFileMerger::MergeRecursive(TDirectory *target, TList *sourcelist, Int_t
                   TDirectory *ndir = nextsource->GetDirectory(dirpath);
                   // For consistency (and persformance), we reset the MustCleanup be also for those
                   // 'key' retrieved indirectly.
-                  ndir->ResetBit(kMustCleanup);
-                  delete ndir;
+                  if (ndir) {
+		    ndir->ResetBit(kMustCleanup);
```
Comment (pcanal) [inline io/io/src/TFileMerger.cxx:780]: Could you replace the ```<tab>``` characters by spaces?  Thanks.

**Comment by pcanal** [comment]
Could the git log be augmented to include the short description of the issues (div by zero, null pointer de-reference) [Without it the commit can not be skimmed without looking at the diff, i.e. git log vs git show].  Thanks.

**Comment by couet** [comment]
@ktf 
I think this PR can be removed right ? the patch in THistPainter.cxx has been already apply ?

**Comment by ktf** [comment]
Wasn't that for master?

**Comment by couet** [comment]
Oh, yes you are right this one is for 6.20.
Thanks. I'll merge.

**Pull request merged**